protocol/x/listing/types/msg_upgrade_isolated_perpetual_to_cross.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	fmt "fmt"
+
+	errorsmod "cosmossdk.io/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ sdk.Msg = &MsgUpgradeIsolatedPerpetualToCross{}
+
+func (msg *MsgUpgradeIsolatedPerpetualToCross) GetSigners() []sdk.AccAddress {
+	addr, _ := sdk.AccAddressFromBech32(msg.Authority)
+	return []sdk.AccAddress{addr}
+}
+
+func (msg *MsgUpgradeIsolatedPerpetualToCross) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return errorsmod.Wrap(
+			ErrInvalidAuthority,
+			fmt.Sprintf(
+				"authority '%s' must be a valid bech32 address, but got error '%v'",
+				msg.Authority,
+				err.Error(),
+			),
+		)
+	}
+	if msg.PerpetualId == 0 {
```

Comment:
what exactly is this check?

Reply:
CodeRabbit suggested: https://github.com/dydxprotocol/v4-chain/pull/2551#discussion_r1842605196

Reply:
perp id 0 is a technically a valid perp id

I think the only check you need to worry about is making sure if the perpetual actually exists and its not already CROSS